src/lib/verifier/prod.ml

```diff
@@ -355,8 +355,12 @@ let with_retry ~logger f =
 let verify_blockchain_snarks {worker; logger} chains =
   with_retry ~logger (fun () ->
       let%bind {connection; _} = !worker in
-      Worker.Connection.run connection ~f:Worker.functions.verify_blockchains
-        ~arg:chains )
+      Deferred.any
+        [ ( after (Time.Span.of_min 3.)
```

Comment:
At the least, this should break out of the `with_retry` loop here, otherwise could retry another 4 times.